cmd/openshift-install/create.go

```diff
@@ -422,6 +423,8 @@ func waitForBootstrapConfigMap(ctx context.Context, client *kubernetes.Clientset
 				timeout = 60 * time.Minute
 			}
 		}
+
+		checkIfAgentCommand(assetStore)
```

Comment:
When testing this PR, I didn't see the warning for wait-for bootstrap-complete. I think it is because the checkIfAgentCommand should be under waitForBootstrapComplete instead of waitForBootstrapConfigMap as it is here.

Reply:
Thanks Richard, I do see it with the bootstrap command (see below). I'm not sure moving it into the waitForBootstrapComplete would gain much, although the message would appear a few seconds we'd have to load the assetStore again.

```
level=info msg=Waiting up to 20m0s (until 9:04AM) for the Kubernetes API at https://api.ostest.test.metalkube.org:6443...
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.ostest.test.metalkube.org:6443/version": dial tcp 192.168.111.5:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.ostest.test.metalkube.org:6443/version": dial tcp 192.168.111.5:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.ostest.test.metalkube.org:6443/version": dial tcp 192.168.111.5:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.ostest.test.metalkube.org:6443/version": dial tcp 192.168.111.5:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.ostest.test.metalkube.org:6443/version": dial tcp 192.168.111.5:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.ostest.test.metalkube.org:6443/version": dial tcp 192.168.111.5:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.ostest.test.metalkube.org:6443/version": dial tcp 192.168.111.5:6443: connect: no route to host
level=debug msg=Still waiting for the Kubernetes API: Get "https://api.ostest.test.metalkube.org:6443/version": dial tcp 192.168.111.5:6443: connect: connection refused
level=info msg=API v1.25.4+77bec7a up
level=debug msg=Loading Install Config...
level=debug msg=  Loading SSH Key...
level=debug msg=  Loading Base Domain...
level=debug msg=    Loading Platform...
level=debug msg=  Loading Cluster Name...
level=debug msg=    Loading Base Domain...
level=debug msg=    Loading Platform...
level=debug msg=  Loading Networking...
level=debug msg=    Loading Platform...
level=debug msg=  Loading Pull Secret...
level=debug msg=  Loading Platform...
level=debug msg=Loading Agent Config...
level=debug msg=Using Agent Config loaded from state file
level=warning msg=An agent configuration was detected but this command is not the agent wait-for command
level=info msg=Waiting up to 30m0s (until 9:20AM) for bootstrapping to complete...


```

Reply:
The part before the k8s API comes up is the part where the `agent wait-for` command adds value - it'll (maybe) tell you _why_ the API didn't come up. So I think we need to either load the assets twice or refactor to load the assets earlier.

I've noticed that in IPI if you just do `create cluster` it ends up loading the assets multiple times (before bootstrapping, waiting for bootstrap-complete, and waiting for cluster-complete), so maybe a refactoring could even reduce that as well.